src/watershed.jl

```diff
@@ -1,14 +1,18 @@
 import Base.isless
 
-struct PixelKey{CT}
+struct PixelKey{CT, N}
     val::CT
     time_step::Int
+    source::CartesianIndex{N}
 end
-isless(a::PixelKey{T}, b::PixelKey{T}) where {T} = (a.val < b.val) || (a.val == b.val && a.time_step < b.time_step)
+isless(a::PixelKey{T, N}, b::PixelKey{T, N}) where {T, N} = (a.val < b.val) || (a.val == b.val && a.time_step < b.time_step)
```

Comment:
I'm actually unsure why we even add the type parameter at all. Is it bad to potentially compare `PixelKey`s with different `val` types?

Reply:
In my comment I meant adding the type parameter to this method, not to the struct. Meaning it could be defined `isless(a::PixelKey, b::PixelKey) = ...` But if you need the `N` to match you can either reorder the parameters or use `PixelKey{S,N}` and `PixelKey{T,N}`.